network/p2p/throttler.go

```diff
@@ -0,0 +1,92 @@
+// Copyright (C) 2019-2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package p2p
+
+import (
+	"sync"
+	"time"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/utils/timer/mockable"
+)
+
+var _ Throttler = (*SlidingWindowThrottler)(nil)
+
+type Throttler interface {
+	// Handle returns if a message from [nodeID] should be handled or not.
+	Handle(nodeID ids.NodeID) bool
+}
+
+// NewSlidingWindowThrottler returns a new instance of SlidingWindowThrottler
+func NewSlidingWindowThrottler(period time.Duration, limit int) *SlidingWindowThrottler {
```

Comment:
Could we add a comment about how limit works here? (seems it is for 2 periods)

Reply:
Added